parcels/fieldset.py

```diff
@@ -103,6 +103,8 @@ def add_field(self, field, name=None):
         :param name: Name of the :class:`parcels.field.Field` object to be added
         """
         name = field.name if name is None else name
+        if name in [fld.name for fld in self.fields]:
```

Comment:
So, in here, you don't check that all `fld` in `fset.fields` have a different name.
You check that a field that you add with `add_field` has a different name than all `fset.fields`.

That's why it does not conflict with #555 
I don't see why this should crash, so I think it's cool to do it that way. But it would be good to add a comment explaining this

Reply:
I can't comment two other lines, since they are not part of the commit, but this PR should also fix some inconsistent ways of adding NestedFields and SummedFields:

In line 114, we add all fields of a NestedField to the fset. So basically, the test you did with same name for `SummedFields` would not pass for `NestedFields`. So I would remove that line.

In function fset.fields (line 500, with property decorator):
`NestedFields` should be added to fields in the same way as `SummedFields`

Also, this function does list all fields in subfields, but it does not list scalar components of vector fields. Is this what we want?